pheps/phep-0003.md

```diff
@@ -0,0 +1,374 @@
+Discussions-To: https://github.com/heliophysicsPy/standards/pull/29
+Revision: 1
+Status: Draft
+Type: Standards Track
+Content-Type: text/markdown; charset=UTF-8; variant=CommonMark
+Created: 06-Jun-2024
+Post-History: 06-Jun-2024, 11-Jun-2024, 02-Jul-2024, 17-Jul-2024, 23-Jul-2024
+```
+
+# Abstract
+<a name="abstract"></a>
+This PHEP recommends that all projects across the PyHC ecosystem adopt a common time-based policy for support of dependencies, inspired by [SPEC 0](https://scientific-python.org/specs/spec-0000/). 
+Specifically, for Python versions and the upstream Scientific Python packages covered by SPEC 0, it recommends that projects:
+1. Support Python versions for at least **36 months** (3 years) after their initial release.
+2. Support upstream Scientific Python packages for at least **24 months** (2 years) after their initial release.
+3. Adopt support for new versions of these dependencies within **6 months** of their release.
+
+The upstream Scientific Python packages are: `numpy, scipy, matplotlib, pandas, scikit-image, networkx, scikit-learn, xarray, ipython, zarr`.
+
+This policy will replace the current standard [#11](https://github.com/heliophysicsPy/standards/blob/main/standards.md#standards) which simply mandates Python 3 support.
+
+# Motivation
+<a name="motivation"></a>
+The current PyHC standard [#11](https://github.com/heliophysicsPy/standards/blob/main/standards.md#standards), which mandates compatibility with Python 3, is outdated. 
+Python 3 support is virtually universal now, so it would be more beneficial to replace this standard with a policy for how to support new minor Python versions and key upstream dependencies. 
+[SPEC 0](https://scientific-python.org/specs/spec-0000/) provides a structured support timeline that balances stability and progress, essential for software in the heliophysics community. 
+Adopting a similar policy ensures consistency and predictability in support timelines. 
+Additionally, limiting the scope of supported versions is an effective way for packages to limit maintenance burden while promoting interoperability.
+
+# Rationale
+<a name="rationale"></a>
+Following [SPEC 0](https://scientific-python.org/specs/spec-0000/)'s 24/36-month support timeline keeps PyHC in better sync with the broader Scientific Python community, maintaining compatibility with newer Python features and key upstream dependencies, while providing adequate time for package maintainers to adapt. 
+Allowing 6 months to adopt new versions ensures packages stay current with development cycles while providing a reasonable timeframe for testing and integration.
+
+# Specification
+<a name="specification"></a>
+This PHEP refers to feature releases of dependencies (e.g., Python 3.12.0, NumPy 2.0.0; not Python 3.12.1, NumPy 2.0.1). 
+
+This PHEP adopts Scientific Python's [SPEC 0](https://scientific-python.org/specs/spec-0000/) and specifies that all PyHC packages should:
```

Comment:
Rather than "specifies that all PyHC packages should" I'd say "specifies that packages must". I.e. a package that is compliant with this PHEP must support that much. The question of the consequences of noncompliance feels more appropriately dealt with elsewhere (e.g. PHEP4 which I haven't looked at yet :) )

I also think involvement in PyHC is sort of implied in the context.

Reply:
This goes back to my reply above about how we previously decided this policy has to be a "should" not a "must." With regard to PHEP 4, rather than thinking about it as "consequences of noncompliance" I'm thinking about it like "if you follow PHEP 3's recommendation then you are compliant for PHEP 4's sake."

Point taken about PyHC involvement being implied, but saying "PyHC" only adds one word and I like the clarity :)